models/users.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+var Db        = require('../config/database');
+var Session   = require('./sessions');
+var validator = require('validator');
+var mongoose  = require('mongoose');
+
+// Create the schema for a user
+var UserSchema = mongoose.Schema({
+    username : { type: String, required: true },
+    email    : { type: String, required: true },
+    password : { type: String, required: true }
+});
+
+// Creates a collection named users in MongoDB
+var UserMongoModel = Db.model('users', UserSchema);
+
+// Takes username, password, email and saves a new user to MongoDB
+function createUser(username, password, email, callback) {
+    var newUser = new UserMongoModel({
+        username : username,
+        email    : email,
+        password : password
+    });
+
+    // Check if newUser.username is already in UserMongoModel
+    UserMongoModel.find({ username : username }).limit(1).exec(function(err, user) {
```

Comment:
collection.find is apparently a whole lot faster because it returns a cursor instead of a document, so it's more ideal for checking if a document exists... Refer to: https://blog.serverdensity.com/checking-if-a-document-exists-mongodb-slow-findone-vs-find/